lib/sync.coffee

```diff
@@ -166,7 +166,14 @@ module.exports =
 		resin.models.device.isOnline(params.uuid).tap (isOnline) ->
 			throw new Error('Device is not online') if not isOnline
 		.then ->
-			return resin.models.device.get(params.uuid).get('uuid').then(performSync)
+			return resin.models.device.hasDeviceUrl(params.uuid)
+		.then (hasDeviceUrl) ->
+			if hasDeviceUrl
+				return resin.models.device.get(params.uuid).get('uuid').then(performSync)
```

Comment:
The `resin.models.device.get` call is shared in both branches of the conditional, better to reuse it to avoid duplication with something like this:

```coffee
.then (hasDeviceUrl) ->
  return resin.models.device.get(params.uuid)
    .get('uuid')
    .tap (fullUUID) ->
      if not hasDeviceUrl
        return resin.models.device.enableDeviceUrl(fullUUID).delay(2000)
    .then(performSync)
```